tests/unit/test-cmd/test.build.js

```diff
@@ -64,301 +56,271 @@ describe('build', () => {
-          .then((buildResult) => {
-            assert.match(buildResult.extensionPath,
-                         /name_of_the_extension-1\.0\.zip$/);
-            return buildResult.extensionPath;
-          })
-    );
-  });
-
-  it('handles repeating localization keys', () => {
-    return withTempDir(
-      (tmpDir) => {
-        const messageFileName = path.join(tmpDir.path(), 'messages.json');
-        fs.writeFileSync(messageFileName,
-          `{"extensionName": {
-              "message": "example extension",
-              "description": "example description"
-            }
-          }`);
-
-        const manifestWithRepeatingPattern = {
-          name: '__MSG_extensionName__ __MSG_extensionName__',
-          version: '0.0.1',
-        };
+  it('gives the correct name to a localized extension', () => withTempDir(
+    async (tmpDir) => {
+      const buildResult = await build({
+        sourceDir: fixturePath('minimal-localizable-web-ext'),
+        artifactsDir: tmpDir.path(),
+      });
+      assert.match(buildResult.extensionPath,
+                   /name_of_the_extension-1\.0\.zip$/);
+      await buildResult.extensionPath;
+    }
+  ));
 
-        return getDefaultLocalizedName({
-          messageFile: messageFileName,
-          manifestData: manifestWithRepeatingPattern,
-        })
-          .then((result) => {
-            assert.match(result, /example extension example extension/);
-          });
-      }
-    );
-  });
+  it('handles repeating localization keys', () => withTempDir(
+    async (tmpDir) => {
+      const messageFileName = path.join(tmpDir.path(), 'messages.json');
+      fs.writeFileSync(messageFileName,
+        `{"extensionName": {
+            "message": "example extension",
+            "description": "example description"
+          }
+        }`);
+
+      const manifestWithRepeatingPattern = {
+        name: '__MSG_extensionName__ __MSG_extensionName__',
+        version: '0.0.1',
+      };
+
+      const result = await getDefaultLocalizedName({
+        messageFile: messageFileName,
+        manifestData: manifestWithRepeatingPattern,
+      });
+      assert.match(result, /example extension example extension/);
+    }
+  ));
 
-  it('checks locale file for malformed json', () => {
-    return withTempDir(
-      (tmpDir) => {
-        const messageFileName = path.join(tmpDir.path(), 'messages.json');
-        fs.writeFileSync(messageFileName,
-          '{"simulated:" "json syntax error"');
-        return getDefaultLocalizedName({
+  it('checks locale file for malformed json', () => withTempDir(
+    async (tmpDir) => {
+      const messageFileName = path.join(tmpDir.path(), 'messages.json');
+      fs.writeFileSync(messageFileName, '{"simulated:" "json syntax error"');
+      let expectedError;
+      try {
+        await getDefaultLocalizedName({
           messageFile: messageFileName,
           manifestData: manifestWithoutApps,
-        })
-          .then(makeSureItFails())
```

Comment:
we can also map the current test strategy by using `makeSureItFails` inside the try block to make sure that the catch block is always executed and then put the assertions directly inside the catch block:

```
try {
  await getDefaultLocalizedName(...);
  makeSureItFails();
} catch (error) {
  assert.instanceOf(error, UsageError);
  ...
}
```